bears/python/YapfBear.py

```diff
@@ -0,0 +1,161 @@
+            Inserts a blank line before a ``def`` or ``class`` immediately
+            nested within another ``def`` or ``class``.
+        :param coalesce_brackets:
+            Does not allow splitting consecutive brackets. Only relevant when
+            ``dedent_closing_brackets`` is set.
+        :param continuation_tab_width:
+            Indent width used for line continuations.
+        :param dedent_closing_brackets:
+            Puts closing brackets on a separate line, dedented, if the bracketed
+            expression can't fit in a single line. Applies to all kinds of
+            brackets, including function definitions and calls.
+        :param indent_dictionary_value:
+            Indent the dictionary value if it cannot fit on the same line as
+            the dictionary key.
+        :param join_multiple_lines:
+            Join short lines into one line.
+        :param spaces_around_power_operator:
+            Set to ``True`` to prefer using spaces around ``**``.
+        :param spaces_before_comment:
+            The number of spaces required before a trailing comment.
+        :param space_between_ending_comma_and_closing_bracket:
+            Inserts a space between the ending comma and closing bracket of a
+            list, etc.
+        :param split_arguments_when_comma_terminated:
+            Splits before arguments if the argument list is terminated by a
+            comma.
+        :param split_before_bitwise_operator:
+            Set to ``True`` to prefer splitting before ``&``, ``|`` or ``^``
+            rather than after.
+        :param split_before_first_argument:
+            If an argument / parameter list is going to be split, then split
+            before the first argument.
+        :param split_before_logical_operator:
+            Set to ``True`` to prefer splitting before ``and`` or ``or`` rather
+            than after.
+        :param split_before_named_assigns:
+            Splits named assignments onto individual lines.
+        :param split_penalty_after_opening_bracket:
+            The penalty for splitting right after the opening bracket.
+        :param split_penalty_after_unary_operator:
+            The penalty for splitting the line after a unary operator.
+        :param split_penalty_before_if_expression:
+            The penalty for splitting right before an if expression.
+        :param split_penalty_bitwise_operator:
+            The penalty of splitting the line around the ``&``, ``|``, and ``^``
+            operators.
+        :param split_penalty_import_names:
+            The penalty of splitting a list of ``import as`` names.
+        :param split_penalty_logical_operator:
+            The penalty of splitting the line around the ``and`` and ``or``
+            operators.
+        :param use_tabs:
+            Use the Tab character for indentation.
+        """
+
+        options = '[style]\nindent_width = ' + str(tab_width)
```

Comment:
i just learned something new :)
thanks Lasse :)

2016-07-15 20:44 GMT+01:00 Lasse Schuirmann <notifications@github.com>:

> In bears/python/YapfBear.py
> <https://github.com/coala-analyzer/coala-bears/pull/578#discussion_r71030497>
> :
>
> > +            The penalty for splitting the line after a unary operator.
> > +        :param split_penalty_before_if_expression:
> > +            The penalty for splitting right before an if expression.
> > +        :param split_penalty_bitwise_operator:
> > +            The penalty of splitting the line around the ``&``, ``|``, and ``^``
> > +            operators.
> > +        :param split_penalty_import_names:
> > +            The penalty of splitting a list of ``import as`` names.
> > +        :param split_penalty_logical_operator:
> > +            The penalty of splitting the line around the ``and`` and ``or``
> > +            operators.
> > +        :param use_tabs:
> > +            Use the Tab character for indentation.
> > +        """
> > +
> > +        options = '[style]\nindent_width = ' + str(tab_width)
>
> hey, I think we could do some python magic here :) We can do:
>
> options = """[style]indent_width = {tab_width}column_limit = {max_line_length}""".format(**locals())
>
> locals() will return a dictionary containing all local variables in this
> scope, this way the template is way better readable :)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/coala-analyzer/coala-bears/pull/578/files/1cac7bdde4d1ce39025b71131f5710239be44e59#r71030497>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AJ9xovuyXcslWrRIIG71no2OtEEu9wf-ks5qV-MOgaJpZM4JMTJw>
> .
>